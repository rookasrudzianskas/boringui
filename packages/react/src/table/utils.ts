import React, {ReactNode} from "react";
import {TableCollection} from "@react-types/table";

export const hasPaginationChild = (
  children: ReactNode | undefined,
  child: React.ElementType,
): {hasPagination: boolean; rowsPerPage: number} => {
  const pagination = React.Children.toArray(children).find(
    (item) => React.isValidElement(item) && item.type === child,
  );

  if (React.isValidElement(pagination)) {
    return {
      hasPagination: true,
      rowsPerPage: pagination?.props?.rowsPerPage || 0,
    };
  }

  return {hasPagination: false, rowsPerPage: 0};
};

export const isInfinityScroll = (collection: TableCollection<unknown>) => {
  const bodyProps = collection.body?.props;

  return bodyProps?.loadingState !== undefined && typeof bodyProps?.onLoadMore === "function";
};
