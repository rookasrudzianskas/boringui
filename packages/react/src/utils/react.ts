import * as React from "react";

export function getValidChildren(children: React.ReactNode) {
  return React.Children.toArray(children).filter((child) =>
    React.isValidElement(child),
  ) as React.ReactElement[];
}

export function addChildrenClass(children: React.ReactNode, className: string) {
  const validChildren = getValidChildren(children);

  return validChildren.map((child) => {
    return React.cloneElement(child, {
      className: `${child.props.className} ${className}`,
    });
  });
}
