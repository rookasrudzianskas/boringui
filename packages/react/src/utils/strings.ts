export const kebabCase = (text: string) => {
  return text
    ?.replace?.(/([a-z])([A-Z])/g, "$1-$2")
    ?.replace?.(/\s+/g, "-")
    .toLowerCase?.();
};

export const includes = (text?: string, search?: string[]) => {
  return search?.some((item) => text?.includes(item));
};
