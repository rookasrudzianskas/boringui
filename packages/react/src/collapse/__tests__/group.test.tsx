import React from "react";
import {mount, render, ReactWrapper} from "enzyme";
import {updateWrapper} from "tests/utils";

import Collapse from "../index";

const getCollapseElement = (wrapper: ReactWrapper, className: string) => {
  return wrapper.find(className).at(0).find(".nextui-collapse-view").at(0);
};

const expectCollapseToHaveARIAExpanded = (
  wrapper: ReactWrapper,
  className: string,
  value: boolean,
) => {
  expect(getCollapseElement(wrapper, className).props()["aria-expanded"]).toBe(value);
};

describe("Collapse Group", () => {
  it("should render correctly", () => {
    const wrapper = mount(
      <Collapse.Group>
        <Collapse title="title1">content1</Collapse>
        <Collapse title="title2">content2</Collapse>
      </Collapse.Group>,
    );

    expect(wrapper).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should be no errors when children are missing", () => {
    const wrapper = mount(<Collapse.Group />);

    expect(wrapper).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work without accordion", () => {
    const wrapper = render(
      <Collapse.Group accordion={false}>
        <Collapse title="title1">content1</Collapse>
        <Collapse title="title2">content2</Collapse>
      </Collapse.Group>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should work with bordered", () => {
    const wrapper = render(
      <Collapse.Group bordered={true}>
        <Collapse title="title1">content1</Collapse>
        <Collapse title="title2">content2</Collapse>
      </Collapse.Group>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should be display all when without accordion", async () => {
    const wrapper = mount(
      <Collapse.Group accordion={false}>
        <Collapse className="test-collapse-1" title="title1">
          content1
        </Collapse>
        <Collapse className="test-collapse-2" title="title2">
          content2
        </Collapse>
      </Collapse.Group>,
    );

    const view1 = getCollapseElement(wrapper, ".test-collapse-1");
    const view2 = getCollapseElement(wrapper, ".test-collapse-2");

    view1.at(0).simulate("click");
    view2.at(0).simulate("click");

    await updateWrapper(wrapper, 300);

    expectCollapseToHaveARIAExpanded(wrapper, ".test-collapse-1", true);
    expectCollapseToHaveARIAExpanded(wrapper, ".test-collapse-2", true);

    view1.at(0).simulate("click");
    view2.at(0).simulate("click");

    await updateWrapper(wrapper, 300);

    expectCollapseToHaveARIAExpanded(wrapper, ".test-collapse-1", false);
    expectCollapseToHaveARIAExpanded(wrapper, ".test-collapse-2", false);
  });

  it("should be display one when is in accordion mode", async () => {
    const wrapper = mount(
      <Collapse.Group>
        <Collapse className="test-collapse-1" title="title1">
          content1
        </Collapse>
        <Collapse className="test-collapse-2" title="title2">
          content2
        </Collapse>
      </Collapse.Group>,
    );

    const view2 = getCollapseElement(wrapper, ".test-collapse-2");

    view2.at(0).simulate("click");
    await updateWrapper(wrapper, 300);

    expectCollapseToHaveARIAExpanded(wrapper, ".test-collapse-1", false);
    expectCollapseToHaveARIAExpanded(wrapper, ".test-collapse-2", true);
  });

  it("should trigger event when collapse changed", () => {
    let index = 0;
    let value = false;

    const callback = jest.fn().mockImplementation((idx: number | undefined, val: boolean) => {
      index = idx || 0;
      value = val;
    });

    const wrapper = mount(
      <Collapse.Group onChange={callback}>
        <Collapse title="title1">content1</Collapse>
        <Collapse title="title2">content2</Collapse>
        <Collapse className="test-collapse" title="title2">
          content3
        </Collapse>
      </Collapse.Group>,
    );
    const view = getCollapseElement(wrapper, ".test-collapse");

    view.at(0).simulate("click");

    expect(callback).toHaveBeenCalled();
    expect(index).toBe(3);
    expect(value).toBe(true);
  });
});
