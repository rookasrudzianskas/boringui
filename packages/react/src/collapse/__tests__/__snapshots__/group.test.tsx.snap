// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Group should be no errors when children are missing 1`] = `ReactWrapper {}`;

exports[`Collapse Group should render correctly 1`] = `ReactWrapper {}`;

exports[`Collapse Group should work with bordered 1`] = `
LoadedCheerio {
  "0": <div
    class="nextui-c-dWWWSv nextui-c-dWWWSv-eaqZHc-borderWeight-light nextui-c-dWWWSv-jXueSE-bordered-true"
  >
    <div
      class="nextui-c-hPzDAx nextui-c-huiNHE nextui-c-hPzDAx-eLdNZT-borderWeight-light nextui-c-hPzDAx-jBQhgJ-divider-true nextui-c-hPzDAx-kMkCGX-animated-true nextui-c-hPzDAx-TDdeu-visible-false nextui-collapse nextui-collapse--closed"
      data-state="closed"
      tabindex="-1"
    >
      <div
        aria-controls="react-aria2612136440-10"
        aria-disabled="false"
        aria-expanded="false"
        class="nextui-c-lfcDHB nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-collapse-view"
        data-state="closed"
        id="react-aria2612136440-9"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-collapse-title-container"
        >
          <div
            class="nextui-collapse-title-content"
          >
            <h3
              class="nextui-collapse-title"
            >
              title1
            </h3>
          </div>
          <div
            class="nextui-collapse-title-content-right"
          >
            <svg
              class="nextui-c-fUXRSc nextui-collapse-icon"
              fill="none"
              focusable="false"
              height="20"
              role="presentation"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="nextui-c-dJMBOt nextui-c-dJMBOt-ielLSBX-css"
      >
        <div
          class="nextui-expand-content"
        >
          <div
            aria-labelledby="react-aria2612136440-9"
            class="nextui-c-fMAxGR nextui-collapse-content"
            id="react-aria2612136440-10"
            role="region"
            tabindex="-1"
          >
            content1
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-hPzDAx nextui-c-huiNHE nextui-c-hPzDAx-eLdNZT-borderWeight-light nextui-c-hPzDAx-jBQhgJ-divider-true nextui-c-hPzDAx-kMkCGX-animated-true nextui-c-hPzDAx-TDdeu-visible-false nextui-collapse nextui-collapse--closed"
      data-state="closed"
      tabindex="-1"
    >
      <div
        aria-controls="react-aria2612136440-12"
        aria-disabled="false"
        aria-expanded="false"
        class="nextui-c-lfcDHB nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-collapse-view"
        data-state="closed"
        id="react-aria2612136440-11"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-collapse-title-container"
        >
          <div
            class="nextui-collapse-title-content"
          >
            <h3
              class="nextui-collapse-title"
            >
              title2
            </h3>
          </div>
          <div
            class="nextui-collapse-title-content-right"
          >
            <svg
              class="nextui-c-fUXRSc nextui-collapse-icon"
              fill="none"
              focusable="false"
              height="20"
              role="presentation"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="nextui-c-dJMBOt nextui-c-dJMBOt-ielLSBX-css"
      >
        <div
          class="nextui-expand-content"
        >
          <div
            aria-labelledby="react-aria2612136440-11"
            class="nextui-c-fMAxGR nextui-collapse-content"
            id="react-aria2612136440-12"
            role="region"
            tabindex="-1"
          >
            content2
          </div>
        </div>
      </div>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Collapse Group should work without accordion 1`] = `
LoadedCheerio {
  "0": <div
    class="nextui-c-dWWWSv nextui-c-dWWWSv-eaqZHc-borderWeight-light"
  >
    <div
      class="nextui-c-hPzDAx nextui-c-huiNHE nextui-c-hPzDAx-eLdNZT-borderWeight-light nextui-c-hPzDAx-jBQhgJ-divider-true nextui-c-hPzDAx-kMkCGX-animated-true nextui-c-hPzDAx-TDdeu-visible-false nextui-collapse nextui-collapse--closed"
      data-state="closed"
      tabindex="-1"
    >
      <div
        aria-controls="react-aria2612136440-6"
        aria-disabled="false"
        aria-expanded="false"
        class="nextui-c-lfcDHB nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-collapse-view"
        data-state="closed"
        id="react-aria2612136440-5"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-collapse-title-container"
        >
          <div
            class="nextui-collapse-title-content"
          >
            <h3
              class="nextui-collapse-title"
            >
              title1
            </h3>
          </div>
          <div
            class="nextui-collapse-title-content-right"
          >
            <svg
              class="nextui-c-fUXRSc nextui-collapse-icon"
              fill="none"
              focusable="false"
              height="20"
              role="presentation"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="nextui-c-dJMBOt nextui-c-dJMBOt-ielLSBX-css"
      >
        <div
          class="nextui-expand-content"
        >
          <div
            aria-labelledby="react-aria2612136440-5"
            class="nextui-c-fMAxGR nextui-collapse-content"
            id="react-aria2612136440-6"
            role="region"
            tabindex="-1"
          >
            content1
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-hPzDAx nextui-c-huiNHE nextui-c-hPzDAx-eLdNZT-borderWeight-light nextui-c-hPzDAx-jBQhgJ-divider-true nextui-c-hPzDAx-kMkCGX-animated-true nextui-c-hPzDAx-TDdeu-visible-false nextui-collapse nextui-collapse--closed"
      data-state="closed"
      tabindex="-1"
    >
      <div
        aria-controls="react-aria2612136440-8"
        aria-disabled="false"
        aria-expanded="false"
        class="nextui-c-lfcDHB nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-collapse-view"
        data-state="closed"
        id="react-aria2612136440-7"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-collapse-title-container"
        >
          <div
            class="nextui-collapse-title-content"
          >
            <h3
              class="nextui-collapse-title"
            >
              title2
            </h3>
          </div>
          <div
            class="nextui-collapse-title-content-right"
          >
            <svg
              class="nextui-c-fUXRSc nextui-collapse-icon"
              fill="none"
              focusable="false"
              height="20"
              role="presentation"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="nextui-c-dJMBOt nextui-c-dJMBOt-ielLSBX-css"
      >
        <div
          class="nextui-expand-content"
        >
          <div
            aria-labelledby="react-aria2612136440-7"
            class="nextui-c-fMAxGR nextui-collapse-content"
            id="react-aria2612136440-8"
            role="region"
            tabindex="-1"
          >
            content2
          </div>
        </div>
      </div>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
